aptos-move/block-executor/src/scheduler.rs

```diff
@@ -247,15 +247,16 @@ impl Scheduler {
                     ))
                 })
                 .collect(),
-            commit_state: CachePadded::new(Mutex::new((0, 0))),
-            execution_idx: AtomicU32::new(0),
-            validation_idx: AtomicU64::new(0),
+            commit_state: CachePadded::new(Mutex::new((first_index, 0))),
+            execution_idx: AtomicU32::new(first_index),
+            validation_idx: AtomicU64::new(first_index as u64),
             done_marker: CachePadded::new(AtomicBool::new(false)),
         }
     }
 
-    pub fn num_txns(&self) -> TxnIndex {
-        self.num_txns
+    pub fn is_last_index(&self, index: TxnIndex) -> bool {
```

Comment:
This also looks confusing. Might be better to change how you define `end_index`.

Reply:
related to [this](https://github.com/aptos-labs/aptos-core/pull/8596#discussion_r1231457484)? let's discuss there.